apps/payments/next/app/[locale]/[offeringId]/[interval]/new/route.ts

```diff
@@ -2,36 +2,55 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { setupCartAction } from '@fxa/payments/ui/actions';
 import { auth } from 'apps/payments/next/auth';
 import { headers } from 'next/headers';
 import { redirect } from 'next/navigation';
+import { NextRequest } from 'next/server';
+import { setupCartAction } from '@fxa/payments/ui/actions';
+import { CartEligibilityStatus } from '@fxa/shared/db/mysql/account';
+
+export const dynamic = 'force-dynamic';
 
 interface CheckoutNewParams {
   locale: string;
   offeringId: string;
   interval: string;
 }
 
-export default async function CheckoutNew({
-  params,
-  searchParams,
-}: {
-  params: CheckoutNewParams;
-  searchParams: Record<string, string>;
-}) {
+export async function GET(
+  request: NextRequest,
+  {
+    params,
+  }: {
+    params: CheckoutNewParams;
+  }
+) {
   const { offeringId, interval } = params;
-  const { coupon } = searchParams;
+  const redirectToUrl = request.nextUrl.clone();
+  const searchParams = request.nextUrl.searchParams;
+  const coupon = searchParams.get('coupon') || undefined;
   const session = await auth();
 
   const fxaUid = session?.user?.id;
   const ip = (headers().get('x-forwarded-for') ?? '127.0.0.1').split(',')[0];
 
+  let cartStatus = 'checkout';
   let cartId: string;
   try {
-    cartId = (
-      await setupCartAction(interval, offeringId, undefined, coupon, fxaUid, ip)
-    ).id;
+    const currentCart = await setupCartAction(
+      interval,
+      offeringId,
+      undefined,
+      coupon,
+      fxaUid,
+      ip
+    );
+    cartId = currentCart.id;
+
+    const cartEligibilityStatus = currentCart.eligibilityStatus;
+    if (cartEligibilityStatus === CartEligibilityStatus.UPGRADE) {
+      cartStatus = 'upgrade';
+    }
```

Comment:
This (and the `let cartStatus = 'checkout'` line above it can be simplified using ternary assignment. Additionally, because of how similar it is to `cart.state` that's used elsewhere, can the name be changed to `elegibilityStatus`?
```suggestion
    const elegibilityStatus = currentCart.elegibilityStatus === CartEligibilityStatus.UPGRADE 
        ? 'upgrade'
        : 'checkout';
```